tonic_validate/validate_scorer.py

```diff
@@ -14,50 +17,58 @@ def __init__(
         self.metrics = metrics
         self.model_evaluator = model_evaluator
 
-    def score_run(self, responses: List[LLMResponse]) -> Run:
+    def _score_item_rundata(self, response: LLMResponse) -> RunData:
+        scores: dict[str, float] = {}
+        # We cache per response, so we need to create a new OpenAIService
+        openai_service = OpenAIService(self.model_evaluator)
+        for metric in self.metrics:
+            score = metric.score(response, openai_service)
+            scores[metric.name] = score
+
+        benchmark_item = response.benchmark_item
+        return RunData(
+            scores,
+            benchmark_item.question,
+            benchmark_item.answer,
+            response.llm_answer,
+            response.llm_context_list,
+        )
+
+    def score_run(self, responses: List[LLMResponse], parallelism=2) -> Run:
```

Comment:
By default we should have `parallelism=1`. That way we leave it up to the user to use parallelism, and then the user know their using parallelism. This is in case using parallelism causes any problems, the user will be aware that they are using it.